### PR TITLE
 fix: When exporting route, whether the parameter is empty or not

### PR DESCRIPTION
**Pull request by Jaycean**
Please answer these questions before submitting a pull request

- Why submit this pull request?
- [x] Bugfix
- [ ] New feature provided
- [ ] Improve performance
- [ ] Backport patches

- Related issues
#1383 
___
### Bugfix
- When exporting route data, the current logic will export empty swagger data when the route data is null or there is no route data. Whether to add judgment will directly return an error.

- How to fix?
When null characters are passed in, routeid is not passed in, or all routeid data is exported, there is no data in the database, and an error should be reported directly.

### CODE REVIEW

File: api/internal/handler/data_loader/route_export_test.go [hunk 1]
```diff
@@ -1903,10 +1904,66 @@ func TestExportRoutesSameURI(t *testing.T) {
...
+	mStore.On("Get", mock.Anything).Run(func(args mock.Arguments) {
+	}).Return(route, nil)
+
+	h := Handler{routeStore: mStore}
+	exportInput := &ExportInput{}
+	exportInput.IDs = ""
+	ctx := droplet.NewContext()
+	ctx.SetInput(exportInput)
+	_, err1 := h.ExportRoutes(ctx)
+	assert.Equal(t, errors.New("Parameter IDs cannot be empty"), err1)
```
Comment (tokers) [inline api/internal/handler/data_loader/route_export_test.go:1936]: What about pre defining error object and reusing them here, if the description changed, we have to change it here.

```go
var ErrEmptyID = errors.New("empty id")
```

Comment (Jaycean) [reply api/internal/handler/data_loader/route_export_test.go:1936]: I personally think that this return error is used by this test alone and only once. I don't think it's necessary to declare a new variable

File: api/test/e2e/route_export_test.go [hunk 2]
```diff
@@ -26,6 +26,24 @@ import (
 	"github.com/stretchr/testify/assert"
 )
 
+func TestExport_Route_Data_Empty(t *testing.T) {
+	tests := []HttpTestCase{
```
Comment (liuxiran) [inline api/test/e2e/route_export_test.go:30]: is it necessary to confirm whether the route list is empty?

Comment (Jaycean) [reply api/test/e2e/route_export_test.go:30]: I think this test returns an error when the route list is empty. Before this test, make sure whether the route list is empty will be repeated.

File: api/internal/handler/data_loader/route_export_test.go [hunk 3]
```diff
@@ -1903,10 +1904,66 @@ func TestExportRoutesSameURI(t *testing.T) {
...
+			},
+			"type": "roundrobin"
+		}
+}`
+	var route *entity.Route
+	err := json.Unmarshal([]byte(r1), &route)
+	assert.Nil(t, err)
+	mStore := &store.MockInterface{}
+	mStore.On("Get", mock.Anything).Run(func(args mock.Arguments) {
+	}).Return(route, nil)
```
Comment (starsz) [inline api/internal/handler/data_loader/route_export_test.go:1928]: IMO, mStore.Get will never be called if the input.IDs is "".
So we can remove the mock here.

Comment (Jaycean) [reply api/internal/handler/data_loader/route_export_test.go]: done. Thks for review

**Review by starsz — APPROVED** [review]

**Review by liuxiran — APPROVED** [review]

**Pull request merged**